sources/packages/backend/apps/queue-consumers/src/services/application/application.service.ts

```diff
@@ -58,6 +58,19 @@ export class ApplicationService {
     return updateResult.affected;
   }
 
+  /**
+   * Sub query to validate if an application has assessment already being
+   * processed by the workflow.
+   */
+  private inProgressStatusesExistsQuery = this.studentAssessmentRepo
```

Comment:
Technically I do not see anything wrong but usually, we have been creating similar subqueries always returned from a method. Not a blocker because for this case I do not see a justification to go otherwise but still feel that a method would look better.

Reply:
I initially felt more comfortable towards private method, but look at it I don't have any concerns.